pdfly/update_offsets.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python
+"""
+Updates offsets and lengths in a simple PDF file.
+
+The PDF specification requires that the xref section at the end
+of a PDF file has the correct offsets of the PDF's objects.
+It further requires that the dictionary of a stream object
+contains a /Length-entry giving the length of the encoded stream.
+
+When editing a PDF file using a text-editor (e.g. vim) it is
+elaborate to compute or adjust these offsets and lengths.
+
+This command tries to compute /Length-entries of the stream dictionaries
+and the offsets in the xref-section automatically.
+
+It expects that the PDF file has ASCII encoding only. It may
+use ISO-8859-1 or UTF-8 in its comments.
+Therefore it expects that there a no binary streams.
+It expects that there is one xref-section only.
+It expects that the /Length-entries have default values containing
+enough digits, e.g. /Length 000 when the stream consists of 576 bytes.
+
+EXAMPLE
+   update-offsets -v --encoding UTF-8 issue-297.pdf issue-297.out.pdf
+"""
+
+from collections.abc import Iterable
+from pathlib import Path
+import logging
+import re
+import sys
+
+
+def update_lines(lines_in: Iterable[str], encoding: str) -> Iterable[str]:
+    """Iterates over the lines of a pdf-files and updates offsets.
+
+    The input is expected to be a pdf without binary-sections.
+
+    :param lines_in: An Iterable over the lines including line-breaks.
+    :param encoding: The encoding, e.g. "iso-8859-1" or "UTF-8".
+    :return The output is a list of lines to be written
+            in the given encoding.
+    """
+    logger = logging.getLogger("update_lines")
```

Comment:
As you can see in the code for other commands, `pdfly` uses `rich.console` and not the standard `logging` system to print messages